src/System.Text.RegularExpressions/ref/System.Text.RegularExpressions.cs

```diff
@@ -62,9 +62,22 @@ public partial class MatchCollection : System.Collections.ICollection, System.Co
         void System.Collections.ICollection.CopyTo(System.Array array, int arrayIndex) { }
     }
     public delegate string MatchEvaluator(System.Text.RegularExpressions.Match match);
+    abstract public partial class RegexRunnerFactory
+    {
+        protected RegexRunnerFactory() { }
+        abstract protected internal RegexRunner CreateInstance();
+    }
     public partial class Regex
     {
         public static readonly System.TimeSpan InfiniteMatchTimeout;
+        protected internal string pattern;
```

Comment:
These fields on the desktop framework don't have the `_` property so we are essentially making them match with the desktop when we expose them here. 

As for Caps and CapNames those are properties because of some dependency issues with these APIs between desktop and .NET Core. In particular the desktop exposes those as fields of type Hashtable which is a dependency we don't want to take in the .NET Core implementation so we are planning to expose them as properties of type IDictionary so they can return a Hashtable instance on desktop and a Dictionary<> instance on .NET Core.